emulator/src/hdmi_cec_driver.c

```diff
@@ -0,0 +1,828 @@
+    {
+      //ADD Logical Address only for Sink device
+      gEmulator->cec_hal->emulated_device.logical_address = logicalAddresses;
+      status = HDMI_CEC_IO_SUCCESS;
+    }
+    else
+    {
+      status = HDMI_CEC_IO_INVALID_ARGUMENT;
+    }
+  }
+  else if( gEmulator==NULL || gEmulator->cec_hal == NULL)
+  {
+    status = HDMI_CEC_IO_NOT_OPENED;
+  }
+  EMU_LOG("HdmiCecAddLogicalAddress - status %d",status);
+  return status;
+}
+
+HDMI_CEC_STATUS HdmiCecRemoveLogicalAddress(int handle, int logicalAddresses)
+{
+
+  HDMI_CEC_STATUS status = HDMI_CEC_IO_INVALID_HANDLE;
+
+  if (handle != 0 && gEmulator != NULL && gEmulator->cec_hal != NULL)
+  {
+    //Remove Logical Address only for Sink device
+    if(gEmulator->cec_hal->emulated_device.type == DEVICE_TYPE_TV && logicalAddresses == 0)
+    {
+      if(gEmulator->cec_hal->emulated_device.logical_address == 0x0F)
+      {
+        //Looks like logical address is already removed. 
+        status = HDMI_CEC_IO_NOT_ADDED;
+      }
+      else
+      {
+        //Reset back to 0x0F
+        gEmulator->cec_hal->emulated_device.logical_address = 0x0F;
+        status = HDMI_CEC_IO_SUCCESS;
+      }
+    }
+    else
+    {
+      //Return Invalid argument for source devices or if logical adddress removed is other than 0
+      status = HDMI_CEC_IO_INVALID_ARGUMENT;
+    }
+  }
+  else if( gEmulator==NULL || gEmulator->cec_hal == NULL)
+  {
+    status = HDMI_CEC_IO_NOT_OPENED;
+  }
+  return status;
+}
+
+HDMI_CEC_STATUS HdmiCecGetLogicalAddress(int handle, int* logicalAddress)
+{
+  EMU_LOG("HdmiCecGetLogicalAddress handle[%d]", handle);
+
+  HDMI_CEC_STATUS status = HDMI_CEC_IO_INVALID_HANDLE;
+
+  if (handle != 0 && gEmulator != NULL && gEmulator->cec_hal != NULL)
+  {
+    if(logicalAddress != NULL)
+    {
+      *logicalAddress = gEmulator->cec_hal->emulated_device.logical_address;
+      status = HDMI_CEC_IO_SUCCESS;
+    }
+    else{
+      status = HDMI_CEC_IO_INVALID_ARGUMENT;
+    }
+  }
+  else if( gEmulator==NULL || gEmulator->cec_hal == NULL)
+  {
+    status = HDMI_CEC_IO_NOT_OPENED;
+  }
+  EMU_LOG("HdmiCecGetLogicalAddress - status %d",status);
+  return status;
+}
+
+HDMI_CEC_STATUS HdmiCecSetRxCallback(int handle, HdmiCecRxCallback_t cbfunc, void* data)
+{
+  HDMI_CEC_STATUS status = HDMI_CEC_IO_INVALID_HANDLE;
+
+  if (handle != 0 && gEmulator != NULL && gEmulator->cec_hal != NULL)
+  {
+    gEmulator->cec_hal->callbacks.rx_cb_func = cbfunc;
+    gEmulator->cec_hal->callbacks.rx_cb_data = data;
+    status = HDMI_CEC_IO_SUCCESS;
+  }
+  else if( gEmulator==NULL || gEmulator->cec_hal == NULL)
+  {
+    status = HDMI_CEC_IO_NOT_OPENED;
+  }
+  EMU_LOG("HdmiCecSetRxCallback - status %d",status);
+  return status;
+}
+
+HDMI_CEC_STATUS HdmiCecSetTxCallback(int handle, HdmiCecTxCallback_t cbfunc, void* data)
+{
+  HDMI_CEC_STATUS status = HDMI_CEC_IO_INVALID_HANDLE;
+
+  if (handle != 0 && gEmulator != NULL && gEmulator->cec_hal != NULL)
```

Comment:
obvuscated flow compiler does this, you don't need too, and it will introduce errors in the future.

remove else if flow